gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannel.java

```diff
@@ -315,9 +311,16 @@ private static Get getMetadataRequest(Storage gcs, StorageResourceId resourceId)
 
   private byte[] getFooterContent(long footerOffset, int footerSize) throws IOException {
     ByteBuffer buffer = ByteBuffer.allocate(footerSize);
+    long oldPositionInGrpcStream = this.positionInGrpcStream;
+    long oldBytesToSkipBeforeReading = bytesToSkipBeforeReading;
     this.positionInGrpcStream = footerOffset;
+    this.bytesToSkipBeforeReading = 0;
+
     readFromGCS(buffer, OptionalLong.empty());
-    this.positionInGrpcStream = 0; // reset position to start
+
+    // restore the original values of these variables
+    this.positionInGrpcStream = oldPositionInGrpcStream; // reset position to start
```

Comment:
please fix the comment, we do not reset position to start, we reset to last known position

Reply:
done